src/otx/algorithms/common/adapters/mmcv/utils/config_utils.py

```diff
@@ -698,3 +700,243 @@ def get_data_cfg(config: Union[Config, ConfigDict], subset: str = "train") -> Co
+            elif pipeline["type"] == "MultiScaleFlipAug":
+                img_size = self._get_size_value(pipeline, "multiscaleflipaug")
+                if img_size is not None:
+                    post_img_size = img_size
+                post_img_size = self._estimate_post_img_size(pipeline["transforms"], post_img_size)
+            elif pipeline["type"] == "AutoAugment":
+                post_img_size = self._estimate_post_img_size(pipeline["policies"][0], post_img_size)
+
+        return post_img_size
+
+    @classmethod
+    def _get_size_value(cls, pipeline: Dict, attr: str) -> Union[List[int], None]:
+        for pipeline_attr in cls.PIPELINE_TO_CHANGE[attr]:
+            if pipeline_attr not in pipeline:
+                continue
+            size_val = pipeline[pipeline_attr]
+            if isinstance(size_val, int):
+                return [size_val, size_val]
+            elif isinstance(size_val, tuple):
+                return list(size_val)
+            elif isinstance(size_val, list):
+                return list(size_val[0])
+
+        return None
+
+    def _get_pipelines(self, data_type: str):
+        if "pipeline" in self._data_config[data_type]:
+            return self._data_config[data_type]["pipeline"]
+        if "dataset" in self._data_config[data_type]:
+            return self._data_config[data_type]["dataset"]["pipeline"]
+        raise RuntimeError("Failed to find pipeline.")
+
+    def _set_pipeline_size_vlaue(self, pipeline: Dict, scale: Tuple[Union[int, float], Union[int, float]]):
+        for pipeline_name, pipeline_attrs in self.PIPELINE_TO_CHANGE.items():
+            if pipeline_name in pipeline["type"].lower():
+                for pipeline_attr in pipeline_attrs:
+                    if pipeline_attr in pipeline:
+                        self._set_size_value(pipeline, pipeline_attr, scale)
+
+        if pipeline["type"] == "MultiScaleFlipAug":
+            for sub_pipeline in pipeline["transforms"]:
+                self._set_pipeline_size_vlaue(sub_pipeline, scale)
+
+        if pipeline["type"] == "AutoAugment":
+            for sub_pipelines in pipeline["policies"]:
+                for sub_pipeline in sub_pipelines:
+                    self._set_pipeline_size_vlaue(sub_pipeline, scale)
+
+    @staticmethod
+    def _set_size_value(pipeline: Dict, attr: str, scale: Tuple[Union[int, float], Union[int, float]]):
+        if isinstance(pipeline[attr], int):
+            pipeline[attr] = round(pipeline[attr] * scale[0])
+        elif isinstance(pipeline[attr], list) and isinstance(pipeline[attr][0], tuple):
+            for idx in range(len(pipeline[attr])):
+                pipeline[attr][idx] = (
+                    round(pipeline[attr][idx][0] * scale[0]),
+                    round(pipeline[attr][idx][1] * scale[1]),
+                )
+        else:
+            pipeline[attr] = (round(pipeline[attr][0] * scale[0]), round(pipeline[attr][1] * scale[1]))
+
+
+def get_configurable_input_size(
```

Comment:
How about `get_model_input_size_from_ckpt` or `get_trained_model_input_size`?

Reply:
Actually, it doesn't refer not only model weight but also `input_size_config` user set. If `input_size_config` is set, then model weight is ignored. I make this function just to avoid code duplication in detection and segmentation task.

Reply:
Then, what about `get_configured_input_size()`?

Reply:
I'll chnage it.